client/my-sites/site-settings/form-seo.jsx

```diff
@@ -89,44 +88,57 @@ export const SeoForm = React.createClass( {
 	},
 
 	componentWillReceiveProps( nextProps ) {
-		if ( get( nextProps, 'site.ID' ) !== get( this.props, 'site.ID' ) ) {
-			// Update state when switching sites
-			this.setState( stateForSite( nextProps.site ) );
+		let nextState = stateForSite( nextProps.site );
+
+		// Don't update state for fields the user has edited
+		if ( this.state.dirtyFields ) {
+			nextState = omit( nextState, this.state.dirtyFields );
 		}
+
+		this.setState( nextState );
 	},
 
 	handleMetaChange( event ) {
 		const seoMetaDescription = event.target.value;
 		// Don't allow html tags in the input field
 		const hasHtmlTagError = anyHtmlTag.test( seoMetaDescription );
 
+		let dirtyFields = this.state.dirtyFields || [];
```

Comment:
You never reassign `dirtyFields`, so it could be a constant with destructuring and a default, like this:
```js
const { dirtyFields = [] } = this.state;
```